### ISSUE TITLE
TabFolder snippet violates the Widget#appendTo API (but still works)

### ISSUE THREAD

**Issue by tomsontom** [eclipsesource/tabris-js#757]
The appendTo API is speced to require a Composite to the code shown in the TabFolder-Snippet is not correct (and eg when using the Typescript typings you get an error)

```javascript
var tab = tabris.create("Tab", {
    title: title, // converted to upper-case on Android
    image: {src: image, scale: 2}
  }).appendTo(tabFolder);
```

if you want to align the snippet with your API definition you need to change it to
```javascript
var tab = tabris.create("Tab", {
    title: title, // converted to upper-case on Android
    image: {src: image, scale: 2}
  });
  tabFolder.append(tab);
```
"Interesting" enough the above works ok. The other option naturally is that you modify the API spec make TabFolder inherit from Composite

**Comment by cookieguru** [comment]
A `Composite` can have any type of object appended to it whereas the `TabFolder` can only have `Tab`s appended to it.

**Comment by ralfstx** [comment]
Right. In the implementation, both `parent.append(child)` and `child.appendTo(parent)` are equivalent. However, the `appendTo` spec does not reflect this, so you get a warning in typescript.

**Comment by ralfstx** [comment]
Fixed with commit 58d2d0fab1c9d4a7162ebe3e4ba66f427beebb23
